overlord/snapstate/snapstate.go

```diff
@@ -168,6 +168,23 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	addTask(setupAliases)
 	prev = setupAliases
 
+	// run refresh, revert or install hook
+	if snapst.HasCurrent() {
+		if snapsup.Flags.Revert {
+			revertHook := SetupRevertHook(st, snapsup.Name())
```

Comment:
It seems more intuitive to run the revert hook on the snap revision that is BEING reverted rather than the snap revision that's being reverted TO. Thoughts?